src/MPI.jl

```diff
@@ -103,6 +103,11 @@ function __init__()
     end
 
     @static if Sys.isunix()
+        # dlopen GTL library (if needed) before dlopen'ing the MPI library:
+        # https://github.com/JuliaParallel/MPI.jl/pull/716
+        if ! isnothing(libgtl)
+            Libdl.dlopen(libgtl, Libdl.RTLD_LAZY | Libdl.RTLD_GLOBAL)
+        end
```

Comment:
Hmm: is this redundant with the `lib/MPIPreferences/src/system.jl`?

Reply:
I've changed `libgtl` -> a vector or libraries to preload.  `Libdl.dlopen` also here in case any of the other jll's start to fall into that pattern -- MPItrampoline comes to mind. If the MPItrampoline provider is built against a system MPI that needs preloads, then they wouldn't be loaded in `MPIPreferences.Sytem`. It's possible that some clever linking would prevent this -- but I haven't had time to figure that out. In the meantime, this is me being cautious.